### ISSUE TITLE
React Native 0.62 support

### ISSUE THREAD

**Issue by kesha-antonov** [Kudo/react-native-v8#42]
Hi!
Is it planned?

**Comment by songxiaoliang** [comment]
same, Hope to support soon

**Comment by Kudo** [comment]
Thanks @enahum's help for sending #41 for upgrade.
I've published react-native-v8@0.62.0-patch.0. Please take a look.

**Comment by enahum** [comment]
Awesome @Kudo, will replace our fork with the nom package and report back here in about 10 hours ;)

**Comment by enahum** [comment]
this is working as intended with RN v0.62.0 installed from the npm package.

Thanks @Kudo for the timely PR review/merge and publishing the package

**Comment by codecog** [comment]
Are there any benchmarks for this against Hermes 0.4.0?

**Comment by songxiaoliang** [comment]
@Kudo If I don't need intl, can I just install v8-android-nointl, or do I need to install both react-native-v8 and v8-android-nointl?

**Comment by ammarahm-ed** [comment]
@songxiaoliang react-native-v8 is the base library you need that handles everything. Along with that you need v8-android or v8-android-nointl installed too.

**Comment by songxiaoliang** [comment]
@ammarahm-ed thanks, I have solved this problem through practice. anyway, Currently I am trying the difference between herms and react-native-v8

**Comment by ammarahm-ed** [comment]
@songxiaoliang If you are creating a memory intensive application, v8 is the best choice to go, it has great garbage collection  and memory usage does not peak.

**Comment by enahum** [comment]
@Kudo can you release 0.62.1 on npm? there is no needed changes on the patch, running the script is be enough (ran it locally and is all good, but I can't publish :P)

**Comment by codecog** [comment]
0.62.2 now :)

**Comment by enahum** [comment]
yeah, no changes needed for 0.62.2 either

**Comment by Kudo** [comment]
Published `v0.62.1-patch.0` and `0.62.2-patch.1`, please let me know if there are any further questions.
(Please do not use 0.62.2-patch.0, which is incorrect version from my publish fault.)

**Comment by Kudo** [comment]
@codecog 
> Are there any benchmarks for this against Hermes 0.4.0?

Will find some time to do that. My recent works is about to add inspector support.
Ideally we could do direct debugging and support TurboModules like Hermes.

**Comment by Jeevan-Kishore** [comment]
@Kudo Fantastic work. We were trying to come to a decision based on stats to switch over to V8 from hermes.

Are the benchmarking's available ?

**Comment by Kudo** [comment]
@Jeevan-Kishore @codecog 
I've just updated react-native-js-benchmark to compare JSC 230250, V8 8.0 and Hermes 0.4.1.
The result is here - https://docs.google.com/spreadsheets/d/1XB6fuk-NYZbCDikxQOAJemE-P8cfbsfdXivTeVwcwIk/edit?usp=sharing

**Comment by magrinj** [comment]
@Kudo Thanks for your work and the benchmark, jsc looks better than Hermes and v8 from the results ?
So using up to date jsc with this library https://github.com/react-native-community/jsc-android-should get better performances than v8 ?

**Comment by Kudo** [comment]
I am sorry that I had a mistake to run benchmarks yesterday. (RNV8 not updated).
I've re-examined again with some changes:
1. Use arm64-v8a ABI to test
    - Many devices are arm64 right now.
    - V8 assumes to use less memory due to its Pointer Compression feature.
2. Add V8 JIT variant into the benchmark result.

@magrinj It's hard to say which JS runtime is better. There were some configurations that may impact to the result.
For JSC, it has four tiers of JIT model and we disabled some of them.
For V8, by default I use the Lite Mode (JIT-less mode).
For Hermes, officially they don't have JIT.
You could see the updated Google Sheet, V8 with JIT acts 2x better than others in RenderComponentThroughput test case.

IMO, I would use V8 if apps need heavy JS computing.
Otherwise, Hermes is good enough especially its binary size is really really small.

**Comment by codecog** [comment]
@Kudo is Jit activated on the android package? If not, how would i go about adding this? Appreciate the work you put in here.

**Comment by Kudo** [comment]
@codecog It's my fault not updated the document.
There are packages like v8-android-jit or v8-android-jit-nointl, credit from @sirpy at https://github.com/Kudo/v8-android-buildscripts/pull/9
The integration is pretty much like [this](https://github.com/Kudo/react-native-v8#how-to-reduce-apk-size-)

I will update the document later.

**Comment by codecog** [comment]
@Kudo Thanks for the direction

**Comment by gpawlik** [comment]
Hi! I am experiencing a significant performance degradation (specially with animations) after the upgrade to `react-native@0.62.2` and `react-native-v8@v0.62.2-patch.1` (from `0.61.4` and `0.61.1-patch.0` respectively). Is this something expected?

**Comment by Kudo** [comment]
@gpawlik Could you try RN 0.62.2 with JSC or Hermes?
I was curious if the degradation comes from RN.

On the other hand, please help me to double check the v8-android* version you used before and after upgrade.

**Comment by gpawlik** [comment]
Thanks @Kudo , I have been using `v8-android` `7.5.1` and upgraded to `8.80.1`. I have tried RN `62.2` with hermes and there was a slight improvement comparing with `v8`, and then tried with `jsc` and the performance problem has gone (app is still a little bit slower than on `v8` but acceptable).

**Comment by Kudo** [comment]
@gpawlik I am sorry that getting a little confused from the description:

> the performance problem has gone (app is still a little bit slower than on v8 but acceptable).

It looks like Hermes and JSC has no performance issue on RN 0.62.2 but only V8, right?

**Comment by gpawlik** [comment]
Only JSC has no performance issues, v8 has serious issues when it comes to screen transitions, animations and any touchable interactions. The performance isn't much better on Hermes, but I haven't tried it before 0.62.2 so it is difficult to compare.

**Comment by enahum** [comment]
@gpawlik do you know what may be causing the perf degradation in your code?

In our app, we can use JSC and V8 no problem at all, but if we switch to hermes, we started to see horrible degradation as you described, but the code is so massive that we couldn’t track it down to anything specific.

**Comment by gpawlik** [comment]
@enahum unfortunately I was not able to track down the exact cause of this degradation (similarly, our codebase is massive), but in the coming weeks I might spend more time doing this investigation so I will keep you updated.

**Comment by Kudo** [comment]
The current JSC has lightweight JIT. From my previous benchmark, the JSC has better performance than Hermes or V8 (not the v8-android-jit version).

However, performance highly depends on applications.
Do you use Animated, Animated with native driver or even Reanimated?

**Comment by gpawlik** [comment]
Yes, I use Animated (with `useNativeDriver` specified), although the performance degradation happen equally on screens and components with no animations at all and with simple interactions like button tap, etc.

**Comment by ashishfeels** [comment]
how should i setup in 0.61.4 @Kudo i'm not able to find similarity in build.gradle file
